8252473: [TESTBUG] compiler tests fail with minimal VM: Unrecognized VM option

zzambers opened the pull request:
This change adds ` -XX:-IgnoreUnrecognizedVMOptions` to problematic tests (or `@requires vm.compiler2.enabled` in one case), to prevent failures `Unrecognized VM option` on client VM.

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [ ] Change must be properly reviewed (2 reviews required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer), 1 [Author](https://openjdk.org/bylaws#author))

### Issue
 * [JDK-8252473](https://bugs.openjdk.org/browse/JDK-8252473): [TESTBUG] compiler tests fail with minimal VM: Unrecognized VM option (**Bug** - P4)


### Reviewers
 * [Emanuel Peter](https://openjdk.org/census#epeter) (@eme64 - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/24262/head:pull/24262` \
`$ git checkout pull/24262`

Update a local copy of the PR: \
`$ git checkout pull/24262` \
`$ git pull https://git.openjdk.org/jdk.git pull/24262/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 24262`

View PR using the GUI difftool: \
`$ git pr show -t 24262`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/24262.diff">https://git.openjdk.org/jdk/pull/24262.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/24262#issuecomment-2755284467)
</details>

zzambers commented:
Attached file which shows unrecognized VM options for individual tests.
[unrecognized-options.txt](https://github.com/user-attachments/files/19472912/unrecognized-options.txt)

vnkozlov reviewed (commented):
Can we run some of them with Graal? When no C2 specific flags are used.

eme64 commented:
@zzambers
Generally we want to get away from `@requires vm.compiler2.enabled`, because it means tests are only run on C2 and not other compilers. For example if C2 is disabled and we only have C1. Or only interpreter. Or Graal ...

Why not just add the compile flag `-XX:-IgnoreUnrecognizedVMOptions`? That could be a good alternative for most cases, I think.

zzambers commented:
> Why not just add the compile flag `-XX:-IgnoreUnrecognizedVMOptions`? That could be a good alternative for most cases, I think.

I saw that approach sometimes used as well. (My little probably unfounded concern would be that typos in args could than be silently ignored.)

I can change my approach to use `-XX:-IgnoreUnrecognizedVMOptions` if that approach is preferable.

eme64 commented:
@vnkozlov do you agree that we should use `-XX:-IgnoreUnrecognizedVMOptions`?

@zzambers Graal does not implement all flags, and so you would get the same issue with `Unrecognized VM option`. But it could still be valuable to run the tests with Graal, even if the flags are not doing anything.

eme64 commented:
> My little probably unfounded concern would be that typos in args could than be silently ignored.

That's not completely unfounded, but I think this taking `-XX:-IgnoreUnrecognizedVMOptions` is still preferrable to `@requires vm.compiler2.enabled`.

eme64 commented:
@zzambers Are you still working on this?

eme64 commented:
@zzambers Can you please update the PR description to match your current changes? And then you should copy that PR description to JIRA :)

eme64 commented:
I quickly looked through the changes, and I think it looks ok. It's a little painful to have to add it everywhere though... It also increases the risk of misspelled flags, or using removed flags etc. But I don't have a great alternative solution.

I'll run some internal testing now, please ping me again in 24h :)

zzambers commented:
It seems, I'll have to rebase

zzambers commented:
@eme64 I have rebased my changes on master and fixed conflicts. (caused by integration of [JDK-8350457](https://github.com/openjdk/jdk/pull/24522))

I have also updated PR description.
(I have not changed JIRA as there is no info about fix. Should I add it there?)

eme64 reviewed (commented):
Still looks reasonable. I'll run some testing now, please ping me again in 24h :)

eme64 reviewed (approved):
@zzambers Thanks for doing this work! The tests pass :green_circle:

zzambers commented:
@eme64 thank you for the review
/integrate

eme64 commented:
@zzambers Before we integrate, I'd like to hear if @vnkozlov agrees with the changes too!

/reviewers 2

shipilev commented:
This fell through the cracks, I think?

zzambers commented:
/open

zzambers commented:
/touch
